src/test/java/com/saralein/server/controller/form/FormPutControllerTest.java

```diff
@@ -0,0 +1,117 @@
+package com.saralein.server.controller.form;
+
+import com.saralein.server.Controller.form.FormBody;
+import com.saralein.server.Controller.form.FormModification;
+import com.saralein.server.Controller.form.FormPutController;
+import com.saralein.server.data.FormStore;
+import com.saralein.server.request.Request;
+import com.saralein.server.response.Header;
+import com.saralein.server.response.Response;
+import java.util.HashMap;
+import java.util.LinkedHashMap;
+import static org.junit.Assert.*;
+import org.junit.Before;
+import org.junit.Test;
+
+public class FormPutControllerTest {
+    private FormStore formStore;
+    private FormPutController formPutController;
+
+    @Before
+    public void setUp() {
+        formStore = new FormStore();
+
+        formStore.addData("/form", new LinkedHashMap<String, String>(){{
+            put("My", "Cat");
+            put("More", "Stuff");
+        }});
+
+        FormBody formBody = new FormBody();
+        FormModification formModification = new FormModification();
+        formPutController = new FormPutController(formStore, formBody, formModification);
+    }
+
+    @Test
+    public void updatesExistingDataForSingleData() {
+        Request request = new Request(new HashMap<String, String>(){{
+            put("method", "PUT");
+            put("uri", "/form");
+            put("version", "HTTP/1.1");
+            put("body", "My=Data");
+        }});
+
+        formPutController.createResponse(request);
+        HashMap<String, String> data = formStore.retrieveData("/form");
+
+        assertTrue(data.containsKey("My"));
+        assertEquals("Data", data.get("My"));
+        assertTrue(data.containsKey("More"));
+        assertEquals("Stuff", data.get("More"));
```

Comment:
The `containsKey` assertions seem a little redundant.

Reply:
Yeah, that's a little silly looking at it now.  Will adjust on my current branch.